File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num) {
    int answer = 0;
    
    long long n = num;
    
    int cnt = 0;
    while(n != 1){
        if(n % 2 == 0){
            n /= 2;
        }else{
            n *= 3;
            n++;
        }
        
        cnt++;
    }

    if(cnt > 500)
        answer = -1;
    else
        answer = cnt;
    
    return answer;
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        
        long n = (long)num;
        int cnt = 0;
        
        while(n != 1) {
            if(cnt > 500)
                return -1;
            
            cnt++;
            if(n % 2 == 0)
                n /= 2;
            else 
                n = n * 3 + 1;
        }
        
        return cnt;
        
    }
}